include/zephyr/zephyr.h

```diff
@@ -9,4 +9,6 @@
 
 #include <zephyr/kernel.h>
 
+#warning "<zephyr/zephyr.h> is deprecated, include <zephyr/kernel.h> instead"
```

Comment:
This should say "Set CONFIG_LEGACY_INCLUDE_PATH=y" to disable the warning

Reply:
@gmarull @henrikbrixandersen @nashif 

@rodrigopex suggests using `#message` instead of `#warning`, this might bee a good intermediate solution.

Reply:
is pragma message supported by all compilers?

Reply:
> is pragma message supported by all compilers?

Neither `#warning` nor `#pragma message` are part of the ISO C standard, but `#warning` is so widely supported that we use it without any checks.

Given that `#pragma message` is also universally supported, I see no reason why it should be treated differently.